scipy/signal/_peak_finding.py

```diff
@@ -337,11 +339,13 @@ def peak_prominences(x, peaks, wlen=None):
     >>> plt.vlines(x=peaks, ymin=contour_heights, ymax=x[peaks])
     >>> plt.show()
     """
-    x = np.asarray(x)
+    x = np.asarray(x, dtype=np.float64)
```

Comment:
Might be worth trying `np.ascontiguousarray` here and using the contiguous memoryview syntax in the Cython code (`[::1]`).

Reply:
I wasn't sure about the tradeoffs in that case. If I understand the what I read in [Cython's](http://docs.cython.org/en/latest/src/userguide/memoryviews.html#c-and-fortran-contiguous-memoryviews) and NumPy's documentation as well as [here](https://stackoverflow.com/questions/26998223/what-is-the-difference-between-contiguous-and-non-contiguous-arrays#26999092) correctly this means:

* Indexing of contiguous arrays is slightly (?) faster with the option `[::1]`.
* If the input arrays aren't contiguous there is an time penalty because it needs to be converted / copied? How common is this case? I have a feeling that his case would be rare and this optimization might be worth it.

Are there any other implications if i use `np.ascontiguousarray` and `[::1]` I'm not aware of?

Reply:
`ascontiguousarray` is free if:
 * we're going to make a copy anyway (by changing dtype, or coercing a list)
 * we're already contiguous

Note that in the first case we'd have to do the dtype casting before `ascontiguousarray`, to avoid copying twice.

I'd try it out and see what the benchmarks show. The generated C code with `[::1]` memoryviews should also be slightly different.